cmd/cosign/cli/fulcio/fulcio.go

```diff
@@ -140,13 +169,21 @@ func NewSigner(ctx context.Context, ko options.KeyOpts) (*Signer, error) {
 		}
 	}
 
-	priv, err := cosign.GeneratePrivateKey()
-	if err != nil {
-		return nil, fmt.Errorf("generating cert: %w", err)
-	}
-	signer, err := signature.LoadECDSASignerVerifier(priv, crypto.SHA256)
-	if err != nil {
-		return nil, err
+	var signer signature.SignerVerifier
+	if ko.IssueCertificate && ko.KeyRef != "" {
+		signer, err = key.SignerVerifierFromKeyRef(ctx, ko.KeyRef, ko.PassFunc)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
higher up the call stack as in `cmd/cosign/cli/sign/sign.go` ?

Reply:
I realized that this isn't actually critical to do, because Fulcio will validate public keys also. The main reason we wanted to check imported keys is because they might be used to sign without Fulcio, but anything in this function will go to Fulcio. So this shouldn't be necessary!